app/screens/transactions/Transactions.js

```diff
@@ -173,8 +178,16 @@ class Transactions extends Component<Props, State> {
             <RightSectionLink onClick={this.navigateToExplanation}>Learn more about Spacemesh Transactions</RightSectionLink>
           </RightSection>
         </InnerWrapper>
-      </Wrapper>
-    );
+      </Wrapper>,
+      shouldShowModal && (
+        <AddNewContactModal
+          key="modal"
+          publicWalletAddress={publicWalletAddress}
+          resolve={() => this.setState({ shouldShowModal: false, publicWalletAddress: null })}
```

Comment:
no reason to reset publicWalletAddress, also rename it to publicAddress

Reply:
ok